pinot-query-planner/src/main/java/org/apache/pinot/calcite/sql/fun/PinotOperatorTable.java

```diff
@@ -152,7 +149,17 @@ public static PinotOperatorTable instance() {
       SqlStdOperatorTable.RANK,
       SqlStdOperatorTable.ROW_NUMBER,
 
+      // WINDOW Functions (non-aggregate)
+      SqlStdOperatorTable.LAST_VALUE,
+      SqlStdOperatorTable.FIRST_VALUE,
+      // TODO: Replace these with SqlStdOperatorTable.LEAD and SqlStdOperatorTable.LAG when the function implementations
```

Comment:
Does this mean `IGNORE NULLS` are simply ignored?
I'd suggest using the standard operator, and throw exception when `IGNORE NULLS` is specified but cannot be supported to make the behavior more explicit

Reply:
> Does this mean IGNORE NULLS are simply ignored?

Nope, using `IGNORE NULLS` with `LAG` / `LEAD` will lead to a clear error like this during query planning - `From line 1, column 43 to line 1, column 60: Cannot specify IGNORE NULLS or RESPECT NULLS following 'LAG'`. This is because the custom operators we defined return `false` for `allowsNullTreatment` which means this Calcite validation will fail - https://github.com/apache/calcite/blob/ef1a83f659e8771c65c2541b92d2ef9cc2a05bea/core/src/main/java/org/apache/calcite/sql/SqlNullTreatmentOperator.java#L69-L74.

I'd initially gone with simply throwing a runtime exception in Pinot's `LagValueWindowFunction` / `LeadValueWindowFunction` runtime operators, but the alternative chosen here suggested by @gortiz (defining our own custom `SqlAggFunction`s) is much better because we fail fast during query planning instead of query execution and the error is also clear.